### PR TITLE
refactor(vue): refactor setup-totp and confirm-sign-in components

### PR DESCRIPTION
**Pull request by wlee221**
### Description of changes

Refactors `setup-totp` and `confirm-sign-in` components to use `useAuthenticator` instead of explicit state machine lookup.

### Description of how you validated changes

Existing CI, wrote unit test, and did smoke tests:

<video src="https://github.com/aws-amplify/amplify-ui/assets/43682783/8e96806e-1627-4abb-a540-f86e7038f058" /> | <video src="https://github.com/aws-amplify/amplify-ui/assets/43682783/4dd8c08d-d98a-4caf-8ef5-ede7e3a49369" />
|---|---|

I've also checked sms mfa.

### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/vue/src/composables/__mock__/useAuthenticatorMock.ts [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
-  route: 'idle',
-  socialProviders: [],
-  unverifiedContactMethods: { email: 'test#example.com' },
-  user: {} as AuthenticatorServiceFacade['user'],
-  validationErrors:
-    {} as unknown as AuthenticatorServiceFacade['validationErrors'],
-  totpSecretCode: null,
   initializeMachine: jest.fn(),
+  isPending: false,
+  QRFields: null,
```
Comment (wlee221) [inline packages/vue/src/composables/__mock__/useAuthenticatorMock.ts:11]: Added `QRFields`, and sorted keys.

File: packages/vue/src/components/confirm-sign-in.vue [hunk 2]
```diff
@@ -1,66 +1,61 @@
...
-  getActorState,
   getFormDataFromEvent,
-  SignInState,
   translate,
 } from '@aws-amplify/ui';
-import { computed, ComputedRef, useAttrs } from 'vue';
 
-import { useAuth, useAuthenticator } from '../composables/useAuth';
+import { useAuthenticator } from '../composables/useAuth';
+import { UseAuthenticator } from '../types';
```
Comment (wlee221) [inline packages/vue/src/components/confirm-sign-in.vue:10]: `UseAuthenticator` is `AuthenticatorServiceFacade` plus internal fields such as `QRFields`. React uses the same typing strategy: [L37-L46](https://github.com/aws-amplify/amplify-ui/blob/13d529369749d4771cfd3849b42b9f2c0fd4adfe/packages/react-core/src/Authenticator/hooks/useAuthenticator/types.ts#L37-L46).

File: packages/vue/src/components/setup-totp.vue [hunk 3]
```diff
@@ -1,44 +1,37 @@
...
+import { UseAuthenticator } from '../types';
 import BaseFormFields from './primitives/base-form-fields.vue';
 
 const logger = new Logger('SetupTOTP-logger');
 
-const props = useAuthenticator();
+// `useAuthenticator` is casted for temporary type safety on this file.
+const props = useAuthenticator() as UseAuthenticator;
+const { updateForm, submitForm, toSignIn } = props;
+const { user, totpSecretCode, isPending, error, QRFields } = toRefs(props);
```
Comment (wlee221) [inline packages/vue/src/components/setup-totp.vue:22]:
```suggestion
const { error, isPending, QRFields , totpSecretCode, user} = toRefs(props);
```

File: packages/vue/src/components/setup-totp.vue [hunk 4]
```diff
@@ -1,44 +1,37 @@
...
+import { UseAuthenticator } from '../types';
 import BaseFormFields from './primitives/base-form-fields.vue';
 
 const logger = new Logger('SetupTOTP-logger');
 
-const props = useAuthenticator();
+// `useAuthenticator` is casted for temporary type safety on this file.
+const props = useAuthenticator() as UseAuthenticator;
+const { updateForm, submitForm, toSignIn } = props;
+const { error, isPending, QRFields , totpSecretCode, user} = toRefs(props);
```
Comment (wlee221) [inline packages/vue/src/components/setup-totp.vue:22]:
```suggestion
const { error, isPending, QRFields , totpSecretCode, user } = toRefs(props);
```

File: packages/vue/src/composables/useAuth.ts [hunk 5]
```diff
@@ -72,6 +80,13 @@ export const useAuthenticator = createSharedComposable(() => {
       //@ts-ignore
       useAuthenticatorValue[key] = facade[key];
     }
+
+    // legacy `QRFields` values only used for SetupTOTP page to retrieve issuer information, will be removed in future
+    const qrFields =
+      facade.route === 'setupTOTP' ? getQRFields(state.value) : null;
```
Comment (wlee221) [inline packages/vue/src/composables/useAuth.ts:86]: https://github.com/aws-amplify/amplify-ui/blob/13d529369749d4771cfd3849b42b9f2c0fd4adfe/packages/react-core/src/Authenticator/hooks/useAuthenticator/useAuthenticator.ts#L51-L52

File: packages/vue/src/components/confirm-sign-in.vue [hunk 6]
```diff
@@ -1,66 +1,61 @@
...
 const onBackToSignInClicked = (): void => {
+  // TODO(BREAKING): remove unused emit
+  // istanbul ignore next
   if (attrs?.onBackToSignInClicked) {
     emit('backToSignInClicked');
   } else {
-    send({
-      type: 'SIGN_IN',
-    });
+    toSignIn();
```
Comment (thaddmt) [inline packages/vue/src/components/confirm-sign-in.vue:58]: I know this is out of scope for this PR, but is this method name not a little confusing? I'm definitely not that familiar with authenticator code but I almost feel like this name is missing a verb lol

**Review by thaddmt — APPROVED** [review]
LGTM

File: packages/vue/src/components/confirm-sign-in.vue (hunk 6 above)
Comment (wlee221) [reply packages/vue/src/components/confirm-sign-in.vue:58]: Yep, what it's really trying to say `navigateToSignInRoute`. This is part of public API, so we'll rely on next breaking changes to improve this.

File: packages/vue/src/components/setup-totp.vue [hunk 7]
```diff
@@ -1,44 +1,37 @@
...
+import { UseAuthenticator } from '../types';
 import BaseFormFields from './primitives/base-form-fields.vue';
 
 const logger = new Logger('SetupTOTP-logger');
 
-const props = useAuthenticator();
+// `useAuthenticator` is casted for temporary type safety on this file.
+const props = useAuthenticator() as UseAuthenticator;
+const { updateForm, submitForm, toSignIn } = props;
+const { error, isPending, QRFields , totpSecretCode, user } = toRefs(props);
```
Comment (wlee221) [inline packages/vue/src/components/setup-totp.vue:22]: 🤷 , this is why I shouldn't code directly on GitHub
```suggestion
const { error, isPending, QRFields, totpSecretCode, user } = toRefs(props);
```

File: packages/vue/jest.config.cjs [hunk 8]
```diff
@@ -13,10 +13,10 @@ module.exports = {
   ],
   coverageThreshold: {
     global: {
-      branches: 36,
-      functions: 52,
-      lines: 66,
-      statements: 65,
+      branches: 51,
```
Comment (zchenwei) [inline packages/vue/jest.config.cjs:16]: 🌟

File: packages/vue/src/components/confirm-sign-in.vue [hunk 9]
```diff
@@ -1,66 +1,61 @@
...
+import { UseAuthenticator } from '../types';
 import BaseFormFields from './primitives/base-form-fields.vue';
 
+/** @deprecated Component events are deprecated and not maintained. */
 const emit = defineEmits(['confirmSignInSubmit', 'backToSignInClicked']);
 const attrs = useAttrs();
 
-const { state, send } = useAuth();
+// `useAuthenticator` is casted for temporary type safety on this file.
+const props = useAuthenticator() as UseAuthenticator;
```
Comment (zchenwei) [inline packages/vue/src/components/confirm-sign-in.vue:18]: Can we achieve the same with type annotation?

Comment (zchenwei) [reply packages/vue/src/components/confirm-sign-in.vue:18]: and other places.

Comment (wlee221) [reply packages/vue/src/components/confirm-sign-in.vue]: [44667e1](https://github.com/aws-amplify/amplify-ui/pull/4202/commits/44667e1b5ade482c247e86d86c485a2ffcf27763) Done! And I'll make a follow up PR for making this change to other subcomponents that are outside this PR.

**Review by zchenwei — APPROVED** [review]

**Review by thaddmt — APPROVED** [review]

**Pull request merged**